manim/mobject/number_line.py

```diff
@@ -148,7 +150,10 @@ def __init__(
         self.decimal_number_config = decimal_number_config
         self.numbers_to_exclude = numbers_to_exclude
         self.numbers_to_include = numbers_to_include
-        self.number_scale_value = number_scale_value
+        # Make a small adjustment to number scaling based on the number of decimal places
+        self.number_scale_value = (
+            number_scale_value - decimal_number_config["num_decimal_places"] * 0.075
+        )
```

Comment:
I think the decrease in size should be more pronounced (maybe `1.0` or `0.9`. Although, there are currently some other issues with this implementation. 
- It doesn't affect very large numbers (as you mentioned). 
- Also a large number of decimal places would actually cause the numbers to flip since the scale value would be negative. A minimum scale value should probably be set.
- It overrides a user-inputted scale value.
- (maybe `scale_numbers = False/True` should be a parameter?)
As of now the original purpose of this PR looks good to merge. Adjusting size based on the length of the number could be addressed in a different PR.

Reply:
Yea I agree, makes sense, it might not be trivial to get it right so best done in its' own PR. Do you want me to rollback the part with the scaling or leave it as it is?

Reply:
Roll it back so that it can be done better later.

Reply:
done